apps/chat/src/components/Chat/ModelList.tsx

```diff
@@ -212,9 +377,51 @@ export const ModelList = ({
             disabled={disabled}
             searchTerm={searchTerm}
             isReplayAsIs={isReplayAsIs}
+            openApplicationModal={() => setModalIsOpen(true)}
+            setIsDeleteModalOpen={setIsDeleteModalOpen}
+            setCurrentEntityName={setCurrentEntityName}
+            setCurrentEntityId={setCurrentEntityId}
+            setCurrentEntityReference={setCurrentEntityReference}
+            handlePublish={handlePublish}
           />
         ))}
       </div>
+      {isDeleteModalOpen && (
+        <ConfirmDialog
+          isOpen={isDeleteModalOpen}
+          heading="Confirm deleting application"
+          description="Are you sure you want to delete the application?"
+          confirmLabel="Delete"
+          cancelLabel="Cancel"
+          onClose={(result) => {
+            setIsDeleteModalOpen(false);
+            if (result) {
+              handleDelete();
+            }
+          }}
+        />
+      )}
+      {modalIsOpen && applicationDetail && (
+        <ApplicationDialog
+          isOpen={modalIsOpen}
+          onClose={() => setModalIsOpen(false)}
+          selectedApplication={applicationDetail}
+          currentReference={currentEntityReference}
+          isEdit
+        />
+      )}
+      {currentEntityName && currentEntityId && entity && (
+        <PublishModal
+          entity={entity}
+          type={SharingType.Application}
+          isOpen={isPublishing}
+          onClose={handlePublishClose}
+          publishAction={
+            PublishActions.ADD
+            // isPublishing ? PublishActions.ADD : PublishActions.DELETE
```

Comment:
if this code isn't used in future, please, remove it

Reply:
It should be uncomment when unpublish will be implemented